lib/ansible/modules/packaging/os/bindep.py

```diff
@@ -0,0 +1,199 @@
+       - An explicit list of profiles to filter by
+     required: false
+     default: ['default']
+'''
+
+EXAMPLES = '''
+# Get the list of packages to install on a given host from a file
+- bindep:
+    path: /home/example/bindep.txt
+
+# Get the list of packages to install and install them
+- bindep:
+    path: /home/example/bindep.txt
+  register: ret
+- package:
+    name: '{{ item }}'
+    state: latest
+  with_items: ret.bindep_packages.missing
+
+# Get the list of packages to install on a given host from a list
+- bindep:
+    requirements:
+        - build-essential [platform:dpkg test]
+        - gcc [platform:rpm test]
+        - language-pack-en [platform:ubuntu]
+        - libffi-dev [platform:dpkg test]
+        - libffi-devel [platform:rpm test]
+
+# Get the list of packages to install on a given host for the test profile
+- bindep:
+    requirements:
+        - build-essential [platform:dpkg test]
+        - gcc [platform:rpm test]
+        - language-pack-en [platform:ubuntu]
+        - libffi-dev [platform:dpkg test]
+        - libffi-devel [platform:rpm test]
+    profiles:
+        - test
+'''
+
+
+RETURN = '''
+bindep_packages:
+    description: Dictionary containing information about the requirements.
+    returned: On success.
+    type: complex
+    contains:
+        missing:
+            description: Packages that are missing from the system
+            returned: success
+            type: list
+            sample:
+                - libmysqlclient-dev
+                - libxml2-dev
+        badversion:
+            description: Packages that are installed but at bad versions.
+            returned: success
+            type: list
+            sample:
+                - package: libxml2-dev
+                  version: 2.9.4+dfsg1-2
+                  constraint: ">= 3.0"
+        up_to_date:
+            description: Flag indicating all packages are up to date
+            returned: success
+            type: bool
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import bindep.depends
+    import ometa.runtime
+    HAS_BINDEP = True
+except ImportError:
+    HAS_BINDEP = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(required=False, type='str'),
+            requirements=dict(required=False, type="list"),
+            profiles=dict(required=False, default=['default'], type='list'),
+        ),
+        mutually_exclusive=[['path', 'requirements']],
+        required_one_of=[['path', 'requirements']],
+        supports_check_mode=True
+    )
+
+    if not HAS_BINDEP:
+        module.fail_json(msg='bindep is required for this module')
+    if not hasattr(bindep.depends, 'get_depends'):
+        module.fail_json(
+            msg='bindep is required at version >= 2.2 for this module')
+
+    path = module.params['path']
+    requirements = module.params['requirements']
+    profiles = module.params['profiles']
+
+    try:
```

Comment:
I don't like to frame such big block of code into an exception. Please put the exception only around the command which actually can throw an exception.